src/testing/flow_solver_cases/periodic_cube_flow.h

```diff
@@ -30,6 +30,9 @@ class PeriodicCubeFlow : public FlowSolverCaseBase<dim,nstate>
     /// Destructor
     ~PeriodicCubeFlow() {};
 
+    /// Integrates over the entire domain
+    double integrate_over_domain(DGBase<dim, double> &dg,const std::string integrate_what) const;
```

Comment:
To be consistent with other functions in this class, maybe `integrate_over_domain()` can take a shared_ptr of DGBase instead of a reference.

Reply:
No need. Only pass shared_ptr if the goal is to store it long term to be reused. In this case, it's a function, so passing by reference is the correct choice.